### PR TITLE
build: Rename `PACKAGE_*` variables to `CLIENT_*`

### PR DESCRIPTION
**Pull request by hebasto**
The use of `PACKAGE_NAME` for the project's variable name is problematic, as this name is commonly used in CMake's [interface variables](https://cmake.org/cmake/help/latest/command/find_package.html#config-mode-version-selection). If third-party CMake code handles with scopes improperly, our `PACKAGE_NAME` variable could end up with an unexpected value.

This PR avoids such conflicts by renaming all `PACKAGE_*` variables to `CLIENT_*`.

The code in the master branch works correctly only incidentally. It is definitely broken in https://github.com/bitcoin/bitcoin/pull/30997.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -47,14 +47,14 @@ project(BitcoinCore
   LANGUAGES NONE
 )
 
-set(PACKAGE_VERSION ${PROJECT_VERSION})
+set(BITCOIN_PACKAGE_VERSION ${PROJECT_VERSION})
 if(CLIENT_VERSION_RC GREATER 0)
-  string(APPEND PACKAGE_VERSION "rc${CLIENT_VERSION_RC}")
+  string(APPEND BITCOIN_PACKAGE_VERSION "rc${CLIENT_VERSION_RC}")
```
[the remark this answers is not in the document]
Comment (hebasto) [reply CMakeLists.txt]: Thanks! Updated per your feedback.

**Review by pablomartin4btc — COMMENTED** [review]
ACK c87a9d2e4c0ad7f2653857f34d7e048585f00be2

**Comment by hebasto** [comment]
Friendly ping @ryanofsky @theuni :)

File: CMakeLists.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (fanquake) [reply CMakeLists.txt]: I think if we end up doingthat, we should extend this to renaming other source variables that exist in the PACKAGE_ namespace. This PR as-is is already changing things such that naming becomes more inconsistent in that regard.

[the remark this answers is not in the document]
Comment (hebasto) [reply CMakeLists.txt]: Thanks! Reworked per feedback.

**Comment by hebasto** [comment]
Rebased due to the conflict with the merged https://github.com/bitcoin/bitcoin/pull/26334.

**Comment by fanquake** [comment]
> It is definitely broken in https://github.com/bitcoin/bitcoin/pull/30997.

Shouldn't #30997 be based on this PR then (or at least mention in the PR description it's expected to be broken)?

**Comment by hebasto** [comment]
> > It is definitely broken in #30997.
> 
> Shouldn't #30997 be based on this PR then (or at least mention in the PR description it's expected to be broken)?

[Done](https://github.com/bitcoin/bitcoin/pull/30997#issuecomment-2435213571).

File: cmake/module/GenerateSetupNsi.cmake [hunk 2]
```diff
@@ -5,8 +5,8 @@
 function(generate_setup_nsi)
   set(abs_top_srcdir ${PROJECT_SOURCE_DIR})
   set(abs_top_builddir ${PROJECT_BINARY_DIR})
-  set(PACKAGE_URL ${PROJECT_HOMEPAGE_URL})
-  set(PACKAGE_TARNAME "bitcoin")
+  set(CLIENT_URL ${PROJECT_HOMEPAGE_URL})
+  set(CLIENT_TARNAME "bitcoin")
```
Comment (fanquake) [inline cmake/module/GenerateSetupNsi.cmake:9]: In b75d532ca4703e26f59c7d33a882a6c1295784ed: The `CLIENT_TARNAME` re-naming is incomplete.

Comment (hebasto) [reply cmake/module/GenerateSetupNsi.cmake:9]: Thanks! Fixed.

**Review by pablomartin4btc — COMMENTED** [review]
re-ACK 36a3de0af9a9120aa98a07ca4abbb77ee6e58ef9

Since my previous review: [extended renaming](https://github.com/bitcoin/bitcoin/pull/31042#discussion_r1795498723) to the rest of sources where `PACKAGE_` namespace existed.

**Review by sedited — APPROVED** [review]
ACK 36a3de0af9a9120aa98a07ca4abbb77ee6e58ef9

The last two commits could be squashed together though? I don't think that would complicate the scripted diff.

**Comment by hebasto** [comment]
> The last two commits could be squashed together though? I don't think that would complicate the scripted diff.

Thanks! Squashed.

**Review by sedited — APPROVED** [review]
Re-ACK 2076a0863ce6e7fcd5a1f2c29c38c1305816e57d

**Comment by hebasto** [comment]
Rebased due to the conflict with the merged https://github.com/bitcoin/bitcoin/pull/31130.

**Review by sedited — APPROVED** [review]
Re-ACK 70713303b6399e0627c1960da4fbab48f9cf71e7

**Comment by fanquake** [comment]
Guix Build:
```bash
2df1f3823c8f689140e3a400f0fdc49d4fe0d2bc4d29402d9765bf9821147b28  guix-build-70713303b639/output/aarch64-linux-gnu/SHA256SUMS.part
33a4689e3200b0fbd56d8098b33cf9388baebe641d27dd6dc09813f56ea2abb1  guix-build-70713303b639/output/aarch64-linux-gnu/bitcoin-70713303b639-aarch64-linux-gnu-debug.tar.gz
35f43ddc81853d054544bbe70d7d8461867c026a7b4d3c9ae90ee248bf28c9de  guix-build-70713303b639/output/aarch64-linux-gnu/bitcoin-70713303b639-aarch64-linux-gnu.tar.gz
b87c9dd6990934021ef5a421e46d0159b822b1ddd4457a7bcbad3ab0763eb279  guix-build-70713303b639/output/arm-linux-gnueabihf/SHA256SUMS.part
da4a4a1dcee0d6b0f210b25b8e0320dda1f2249efb90a7677cafcf2f00c1eb56  guix-build-70713303b639/output/arm-linux-gnueabihf/bitcoin-70713303b639-arm-linux-gnueabihf-debug.tar.gz
925b1d6cf6fb2c59a34c74a05053278d88ed7ee84ddeb960f39599835fd31a8e  guix-build-70713303b639/output/arm-linux-gnueabihf/bitcoin-70713303b639-arm-linux-gnueabihf.tar.gz
1a36013e0f1585b5c8ffe4e39fa24638481a2c38d8dafc9aaaece11698236b92  guix-build-70713303b639/output/arm64-apple-darwin/SHA256SUMS.part
9bfcbd2ddc619ccca320424a0520e79f8489c28207093e855bba48322a48f24a  guix-build-70713303b639/output/arm64-apple-darwin/bitcoin-70713303b639-arm64-apple-darwin-unsigned.tar.gz
59c841f7b385bd9645cef3e4b1a5f93b2ef577a276e06418f919180ee23d7e4d  guix-build-70713303b639/output/arm64-apple-darwin/bitcoin-70713303b639-arm64-apple-darwin-unsigned.zip
57019be46ef95ea27ca5a4fb3d29c78782988fbe15ee893aa912b43b3da7dfa0  guix-build-70713303b639/output/arm64-apple-darwin/bitcoin-70713303b639-arm64-apple-darwin.tar.gz
ab6762e153f42dfb37d21d205c14a9ff3fd381b74141a45b9ea38ee129e856e5  guix-build-70713303b639/output/dist-archive/bitcoin-70713303b639.tar.gz
fb9b9719b2769406a6d7d4ad05d737b954c51f374385cb44a336272c3c291736  guix-build-70713303b639/output/powerpc64-linux-gnu/SHA256SUMS.part
d4f2cdf931b7bba6c4d9291ea930659c04fd6a7a3a9682c5268ee96145861c97  guix-build-70713303b639/output/powerpc64-linux-gnu/bitcoin-70713303b639-powerpc64-linux-gnu-debug.tar.gz
f9db6ed69d00b9dfaa72fe7a0658e19b3c841ea0b4ada59a5cd4659dd330d392  guix-build-70713303b639/output/powerpc64-linux-gnu/bitcoin-70713303b639-powerpc64-linux-gnu.tar.gz
8167118fcef084f28831e6bece33eb2061404fa0f7e96d629a95d3f329ba5310  guix-build-70713303b639/output/riscv64-linux-gnu/SHA256SUMS.part
819cd35f0c32a68a545347adb0f6bc6e53c0d243a4bacf37cebf99b400648b35  guix-build-70713303b639/output/riscv64-linux-gnu/bitcoin-70713303b639-riscv64-linux-gnu-debug.tar.gz
521107405bccfd2032356dd06227391d696ebaf1edf87a24330b4d78bb55af49  guix-build-70713303b639/output/riscv64-linux-gnu/bitcoin-70713303b639-riscv64-linux-gnu.tar.gz
1eb6cbf97f5c9a0e66c204b141f6f4780374242b05ac2f55e47fdb571abbafda  guix-build-70713303b639/output/x86_64-apple-darwin/SHA256SUMS.part
509038ecfac467f801f06078868bf52de2918928faea4d2aae9a5784732f976c  guix-build-70713303b639/output/x86_64-apple-darwin/bitcoin-70713303b639-x86_64-apple-darwin-unsigned.tar.gz
213a190c1cfed5f40089b078cef860ffc1ba3a0bedc7e489e89658db40718bef  guix-build-70713303b639/output/x86_64-apple-darwin/bitcoin-70713303b639-x86_64-apple-darwin-unsigned.zip
691f5cd5e4f9659ef15161cc4f30a797c6abd803f60e4eeba36b9d391b23981d  guix-build-70713303b639/output/x86_64-apple-darwin/bitcoin-70713303b639-x86_64-apple-darwin.tar.gz
3c2b462b300659fcf46537c67ceac879f3d5e850c988bb172bb6514dac7701c0  guix-build-70713303b639/output/x86_64-linux-gnu/SHA256SUMS.part
2d9cc4effcc041f87fb6054e56a206694ccf133c32ae62dc4bb6f2107a589beb  guix-build-70713303b639/output/x86_64-linux-gnu/bitcoin-70713303b639-x86_64-linux-gnu-debug.tar.gz
b3d11e2a55ae7cf03f601527816052214596e2270190f89a151eac4e8796c465  guix-build-70713303b639/output/x86_64-linux-gnu/bitcoin-70713303b639-x86_64-linux-gnu.tar.gz
225ec52f051ec7410b29f30b9c36898f330fb5d94267f242b39a18711b4bbe37  guix-build-70713303b639/output/x86_64-w64-mingw32/SHA256SUMS.part
8f022cecbedf22324778983b9d8b7351d24a1d51297e7124f06bf82cc5490bbd  guix-build-70713303b639/output/x86_64-w64-mingw32/bitcoin-70713303b639-win64-debug.zip
0243301dcff8850146901e77b0e58659594fb7108193004c11fa46d5ded439e9  guix-build-70713303b639/output/x86_64-w64-mingw32/bitcoin-70713303b639-win64-setup-unsigned.exe
97938f0053cf1f361817d8bb750461a809f05983ea7d030d61222c53053e91e7  guix-build-70713303b639/output/x86_64-w64-mingw32/bitcoin-70713303b639-win64-unsigned.tar.gz
8e205188361743d291fe4317741a05b7d205c6a7df22c07338a1aaad49159507  guix-build-70713303b639/output/x86_64-w64-mingw32/bitcoin-70713303b639-win64.zip
```

**Review by Armss9936 — APPROVED** [review]

**Pull request merged**